assets/src/blocks/Media/MediaEditor.js

```diff
@@ -0,0 +1,150 @@
+  }
+
+  return <InspectorControls>
+      <PanelBody title={__('Setting', 'planet4-blocks-backend')}>
+        <TextControl
+          label={__('Media URL/ID', 'planet4-blocks-backend')}
+          placeholder={__('Enter URL', 'planet4-blocks-backend')}
+          value={ media_url }
+          onChange={
+            value => {
+              setMediaURL(value)
+              debouncedMediaURLUpdate(value)
+            }
+          }
+          help={__('Can be a YouTube, Vimeo or Soundcloud URL or an mp4, mp3 or wav file URL.', 'planet4-blocks-backend')}
+        />
+
+        <MediaPlaceholder
+          labels={{ title: __('Video poster image [Optional]', 'planet4-blocks-backend'), instructions: __('Applicable for .mp4 video URLs only.', 'planet4-blocks-backend')}}
+          icon="format-image"
+          onSelect={ onSelectImage }
+          onError={ console.log }
+          accept="image/*"
+          allowedTypes={["image"]}
+        />
+      </PanelBody>
+    </InspectorControls>
+  ;
+}
+
+const renderView = (props, toAttribute) => {
+  const { attributes } = props;
+  const { video_title, description, embed_html, media_url, poster_url } = attributes;
+
+  return (
+    <Fragment>
+      <header>
+        <RichText
+          tagName="h2"
+          className="page-section-header"
+          placeholder={__('Enter title', 'planet4-blocks-backend')}
+          value={video_title}
+          onChange={toAttribute('video_title')}
+          keepPlaceholderOnFocus={true}
+          withoutInteractiveFormatting
+          characterLimit={40}
+          multiline="false"
+        />
+      </header>
+      <RichText
+        tagName="p"
+        className="page-section-description"
+        placeholder={__('Enter description', 'planet4-blocks-backend')}
+        value={description}
+        onChange={toAttribute('description')}
+        keepPlaceholderOnFocus={true}
+        withoutInteractiveFormatting
+        characterLimit={200}
+      />
+      {
+        (media_url && !media_url.endsWith('.mp4')) && !embed_html
+        ? <div className="block-edit-mode-warning components-notice is-error">
+            { __( 'The video URL could not be parsed.', 'planet4-blocks-backend' ) }
+          </div>
+        : <Fragment>
+            {
+              media_url && media_url.endsWith('.mp4')
+              ? <MediaElementVideo videoURL={ media_url } videoPoster={ poster_url } />
+              : <MediaEmbedPreview html={ embed_html || null } />
+            }
+          </Fragment>
+      }
+    </Fragment>
+  )
+}
+
+export const MediaEditor = (props) => {
+  const { attributes, setAttributes, isSelected } = props;
+  const { media_url, video_poster_img } = attributes;
+
+  // Assume that a non-URL is a YouTube video ID.
+  const parsedMediaURL = media_url && media_url.indexOf('/') === -1
+    ? `https://www.youtube.com/watch?v=${media_url}`
+    : media_url;
+
+  console.log('media URL', parsedMediaURL);
```

Comment:
Can you remove this logging? We should probably setup `eslint` to disallow these.